### ISSUE TITLE
Wrong error message when using min_size and max_size

### ISSUE THREAD

**Issue by Goltergaul** [dry-rb/dry-validation#236]
I've got this scheme: `BODY_SCHEMA = Dry::Validation.Form do required(:name) { str? & min_size?(3) & max_size?(25) } end`

when i do `BODY_SCHEMA.call({"name" => "aa"})` i get `#<Dry::Validation::Result output={:name=>"aa"} messages={:name=>["size cannot be less than 3", "size cannot be greater than 25"]}>`

but the `size cannot be greater than 25` part is obviously wrong. (I'm using version 0.9.4)

Also i tried using this scheme `BODY_SCHEMA = Dry::Validation.Form do required(:name).filled(:str?, max_size?: 25, min_size?: 3) end`
which gives me `#<Dry::Validation::Result output={:name=>"aa"} messages={}>` which is wrong because it ignores the min_size?

**Comment by timriley** [comment]
The `size cannot be greater than 25` is actually a validation _hint_, a message about a validation rule that wasn't able to be executed on the previous validation run. Hints are definitely still a work-in-progress; read along from https://github.com/dry-rb/dry-validation/issues/225#issuecomment-236345045 if you'd like to learn more about how the hints work now, and how we plan to make them better in the future.

As for your second schema, that definitely seems like a bug to me. I could duplicate that on master. Will leave this for @solnic to take a look at when he gets a chance :)

**Comment by Goltergaul** [comment]
Ah thanks for the info!

**Comment by solnic** [comment]
This is a duplicate of #227 for now do block-based def so:

``` ruby
required(:name).filled { str? & max_size?(25) & min_size?(3) }
```
